src/Cli/dotnet/commands/InstallingWorkloadCommand.cs

```diff
@@ -123,6 +123,13 @@ protected bool TryHandleWorkloadUpdateFromVersion(ITransactionContext context, D
                 _workloadInstaller.UpdateInstallMode(_sdkFeatureBand, true);
             }
 
+            // Delete the current advertising manifest because if we fail to find the right workload version, we want to fail.
+            var advertisingPackagePath = Path.Combine(_userProfileDir, "sdk-advertising", _sdkFeatureBand.ToString(), "microsoft.net.workloads");
+            if (Directory.Exists(advertisingPackagePath))
+            {
+                Directory.Delete(advertisingPackagePath, recursive: true);
+            }
```

Comment:
I think after my refactoring, this is probably not necessary anymore.  Can you verify or explain what the scenario was where this was broken?